Fix sample_distorted_bounding_box where min_object_covered could be None

yongtang opened the pull request:
This fix tried to address the issue raised in #15529 where not providing min_object_covered a value will result in a ValueError. In the docstring, however, min_object_covered has been described as default to 0.1.

The reason for the issue is that when sample_distorted_bounding_box switched to V2, min_object_covered has been changed form an attr to an input. As input could not have a default value, min_object_covered=None will result in an error.

This fix adds the check so that a default value 0.1 will be provided if min_object_covered=None.

This fix fixes #15529.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Fix sample_distorted_bounding_box where min_object_covered could be None

This fix tried to address the issue raised in 15529 where
not providing min_object_covered a value will result in
a ValueError. In the docstring, however, min_object_covered
has been described as default to 0.1.

The reason for the issue is that when sample_distorted_bounding_box
switched to V2, min_object_covered has been changed form an attr
to an input. As input could not have a default value,
min_object_covered=None will result in an error.

This fix adds the check so that a default value 0.1 will be
provided if min_object_covered=None.

This fix fixes 15529.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add test case for min_object_covered=None with sample_distorted_bounding_box.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

rmlarsen commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -1533,6 +1533,8 @@ def sample_distorted_bounding_box(image_size, bounding_boxes, seed=None,
       Provide as input to `tf.image.draw_bounding_boxes`.
   """
   with ops.name_scope(name, 'sample_distorted_bounding_box'):
+    if min_object_covered is None:
```
You should update the default value instead, although that will require an API review, I believe.

rmlarsen replied:
The documentation already stated that the default is 0.1, so I think it would be fine to change it.

yongtang committed:
Move min_object_covered = 0.1 to the args

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Update golden API

with
```
bazel-bin/tensorflow/tools/api/tests/api_compatibility_test
           --update_goldens True
```

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
Thanks @rmlarsen for the review. The PR has been updated.

asimshankar commented:
(Good for API change, thanks!)

rmlarsen commented:
@yongtang please resolve conflict.
